src/main/java/br/com/deveficiente/nossomercadolivreapi/gatewaypagamento/GatewayPagamentoType.java

```diff
@@ -0,0 +1,33 @@
+package br.com.deveficiente.nossomercadolivreapi.gatewaypagamento;
+
+import br.com.deveficiente.nossomercadolivreapi.compra.nova.Compra;
+import org.springframework.util.Assert;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+public enum GatewayPagamentoType {
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

A enum é mais poderosa do que parece :). Exemplo do que você pode fazer:

```
public enum GatewayPagamentoType {
 PAYPAL {
   public String geraUrl(Compra compra,UriComponentsBuilder uriComponentsBuilder){
     //codigo aqui
   },
   PAGSEGURO{...}

   public abstract String geraUrl(Compra compra, UriComponentsBuilder uriComponentsBuilder) ;

 }

}
```

Reply:
Você economiza duas classes :)

Reply:
Refatorei essa parte.